MetaMask/metamask-extension
# Support token detection on Arbitrum, Optimism, Base, and zkSync

## **Description**

Supports token detection on Arbitrum, Optimism, Base, and zkSync

Accompanying PRs to core:
- https://github.com/MetaMask/core/pull/2035
- https://github.com/MetaMask/core/pull/3584

## **Related issues**

- https://github.com/MetaMask/metamask-extension/issues/17697
- https://github.com/MetaMask/metamask-extension/issues/16496

## **Manual testing steps**

1. Load an account on arbitrum, optimism, base, or zkSync that contains ERC20 tokens
2. Enable token autodetection in settings
3. It should detect tokens to be added ("N" new tokens found in this account")
4. Click "Import tokens" to manually import
5. There should be a search box that suggests tokens as you type their symbols

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/3500406/0e81e4a7-331f-4712-9481-317411b158c3

### **After**

https://github.com/MetaMask/metamask-extension/assets/3500406/7721eab9-de55-49a6-a71e-f5124ef6dbc5

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
